ndcube/ndcube.py

```diff
@@ -964,24 +965,91 @@ def _new_instance(self, **kwargs):
     def __neg__(self):
         return self._new_instance(data=-self.data)
 
-    def __add__(self, value):
+    def add(self, value, operation_ignores_mask=True, handle_mask=np.logical_and):
+        """
+        Users are allowed to choose whether they want operation_ignores_mask to be True or False,
+        and are allowed to choose whether they want handle_mask to be AND / OR .
+        """
+        kwargs = {}
+
+        if isinstance(value, NDData) and value.wcs is None:
+            if self.unit is not None and value.unit is not None:
+                value_data = (value.data * value.unit).to_value(self.unit)
+            elif self.unit is None:
+                value_data = value.data
+            else:
+                raise TypeError("Cannot add unitless NDData to a unitful NDCube.")
+
+            # check whether there is a mask.
+            # Neither self nor value has a mask
+            self_unmasked = self.mask is None or self.mask is False or not self.mask.any()
+            value_unmasked = value.mask is None or value.mask is False or not value.mask.any()
+
+            if (self_unmasked and value_unmasked) or operation_ignores_mask is True:
+                # addition
+                kwargs["data"] = self.data + value_data
+
+                # combine the uncertainty;
+                if self.uncertainty is not None and value.uncertainty is not None:
+                    new_uncertainty = self.uncertainty.propagate(
+                        np.add, value, result_data = kwargs["data"], correlation=0
+                    )
+                    kwargs["uncertainty"] = new_uncertainty
+                elif self.uncertainty is not None:
+                    new_uncertainty = self.uncertainty
+                    kwargs["uncertainty"] = new_uncertainty
+                elif value.uncertainty is not None:
+                    new_uncertainty = value.uncertainty
+                else:
+                    new_uncertainty = None
+            else:
+                # TODO
+                # When there is a mask, that is when the two new added parameters (OIM and HM) come into the picture.
+                # Conditional statements to permutate the two different scenarios (when it does not ignore the mask).
+                if operation_ignores_mask is False:
+                    if handle_mask is np.logical_and:
+                        pass
+                    else:
+                        pass
+                else:
```

Comment:
```suggestion
```
For simplicity let's remove this code for now and just focus on the case where there are no masks.  This should make debugging the tests easier.